### DISCUSSION TITLE
Is there a way to display until a valid query completes?

### DISCUSSION THREAD

**Discussion by cfepdan** [Budibase/budibase#6047] [Help]
I have a REST-based query I do that either takes a couple seconds or fails after 10 seconds...

Is there a clever way to display a container that disappears when my query completes successfully? I currently have a container within my Data Provider that shows while my number of rows is zero. This works when the query fails... however, it doesn't show *until* the query fails. I'd like to have something that gets replaced when the query returns successfully.

**Comment by melohagan** [comment]
Hi @cfepdan 

I don't know of a way of doing this, but it does sound useful. I'm assuming you want to add in your own custom loading content e.g. some text that says *Please wait while loading*? Or maybe some buttons you want to be disabled until the data has loaded? 

Being able to define actions for **onLoad** event in data providers could be useful. Then you could update app state to flip a show/hide conditional ui.

**Reply by cfepdan** [reply]
At the moment, I'm hoping to display custom loading content. Like:

"I'll give you more specifics in a moment, but in the meantime, read this information about the data you're going to receive:

Values greater than 100 are good.
Values less than 100 are bad."

And then the specific data shows up, and the "Values greater than 100..." stuff sticks around below the specific data, but the "I'll give you more specifics in a moment" disappears.

**Reply by Havock94** [reply]
I'd find this useful as well, as I was trying to fill a text input based on data loaded on a provider. At the moment, using `{{ Provider.Rows.0.title }}` gives an error, because on page load it has no rows

**Reply by melohagan** [reply]
In your case @Havock94 you could use a Data Provider / Repeater combo, if you are certain only one row will be returned.

Then your binding would be something like: `New Repeater.title`
